src/components/routes/RoutesCalendar.vue

```diff
@@ -0,0 +1,250 @@
+        activeDirection.value === direction
+      );
+    }
+
+    /**
+     * Handles click on route item within a day frame.
+     * It triggers active state on that day.
+     * It controls content of the route-logging dialog panel.
+     * @param param0 { timestamp: Timestamp; direction: TransportDirection }
+     * @returns void
+     */
+    function onClickItem({
+      timestamp,
+      direction,
+    }: {
+      timestamp: Timestamp;
+      direction: TransportDirection;
+    }): void {
+      activeItem.value = timestamp;
+      activeDirection.value = direction;
+    }
+
+    return {
+      activeItem,
+      calendar,
+      locale,
+      monthNameAndYear,
+      routesMap,
+      selectedDate,
+      theme,
+      isActive,
+      onClickItem,
+      onNext,
+      onPrev,
+      onToday,
+    };
+  },
+});
+</script>
+
+<template>
+  <div data-cy="routes-calendar">
+    <!-- Top bar -->
+    <div class="row q-pb-sm q-my-lg items-center gap-8">
+      <!-- Title -->
+      <div
+        class="col-12 col-sm-auto text-h5 text-capitalize text-weight-bold"
+        data-cy="calendar-title"
+      >
+        {{ monthNameAndYear }}
+      </div>
+      <!-- Navigation buttons -->
+      <calendar-navigation
+        @next="onNext"
+        @prev="onPrev"
+        @today="onToday"
+        class="col-12 col-sm"
+      />
+    </div>
+    <!-- Calendar -->
+    <div class="row justify-center q-mt-lg">
+      <q-calendar-month
+        ref="calendar"
+        v-model="selectedDate"
+        animated
+        bordered
+        hoverable
+        no-active-date
+        use-navigation
+        short-weekday-label
+        :locale="locale"
+        :show-month-label="false"
+        :weekdays="[1, 2, 3, 4, 5, 6, 0]"
+        weekday-align="center"
+        :style="theme"
+        date-align="right"
+        date-type="rounded"
+        :day-min-height="100"
+      >
+        <template #day="{ scope: { timestamp } }">
+          <div v-if="!timestamp.future" class="q-my-sm" data-cy="calendar-day">
+            <!-- Route to work -->
+            <calendar-item-display
+              :active="isActive({ timestamp, direction: 'toWork' })"
+              direction="toWork"
+              :day="routesMap[timestamp.date]"
+              :timestamp="timestamp"
+              @item-click="onClickItem"
+              data-cy="calendar-item-display-to-work"
+            />
+            <!-- Route from work -->
+            <calendar-item-display
+              :active="isActive({ timestamp, direction: 'fromWork' })"
+              direction="fromWork"
+              :day="routesMap[timestamp.date]"
+              :timestamp="timestamp"
+              @item-click="onClickItem"
+              data-cy="calendar-item-display-from-work"
+            />
+          </div>
+        </template>
+      </q-calendar-month>
+    </div>
+  </div>
+</template>
+
+<style lang="scss" scoped>
+:deep(.q-calendar-month__day--label) {
```

Comment:
Really need this customization (potentially this code break up with next release)?

Reply:
I did not find a clean api for styling the highlighted (selected) date. Based on design it should have semi-transparent background color and no outline.
I will check again.
If there is no way of styling and we need to follow design closely, we can write tests for it (they should signal breaking change).

Reply:
This was resolved in the team discussion - we are moving forward with default styling.
I removed the styles.